ext/xml/xml.c

```diff
@@ -1012,8 +1012,17 @@ static void php_xml_parser_create_impl(INTERNAL_FUNCTION_PARAMETERS, int ns_supp
 
 	XML_Char *encoding;
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), (ns_support ? "|s!s": "|s!"), &encoding_param, &encoding_param_len, &ns_param, &ns_param_len) == FAILURE) {
-		RETURN_THROWS();
+	if (ns_support) {
+	    ZEND_PARSE_PARAMETERS_START(0, 2)
+	    	Z_PARAM_OPTIONAL
+	        Z_PARAM_STRING_EX(encoding_param, encoding_param_len, 1, 0)
+            Z_PARAM_STRING(ns_param, ns_param_len)
+	    ZEND_PARSE_PARAMETERS_END();
+	} else {
+	    ZEND_PARSE_PARAMETERS_START(0, 2)
+	    	Z_PARAM_OPTIONAL
+	        Z_PARAM_STRING_EX(encoding_param, encoding_param_len, 1, 0)
```

Comment:
Should be the ``Z_PARAM_STRING_OR_NULL`` version, moreover is there a reason for it to not use zend_strings instead of keeping the char* with a separate length.